### PR TITLE
Make navbar sticky and fix 'Check Trends' button functionality

### LINKED ISSUE #81: [FEATURE] Making the Navbar sticky and Fixing "Check Trends" Button.
### Is this a unique feature?

- [X] I have checked "open" AND "closed" issues and this is not a duplicate

### Is your feature request related to a problem/unavailable functionality? Please describe.

I often find it frustrating when navigating the page because the navbar is not sticky, making it hard to access different sections easily. Additionally, the "Check Trends" button previously did not function, which left users unable to scroll directly to the "Repos" section.

### 'Nav.tsx'- (in line 16)
```  <nav className="bg-gray-800 p-4 w-screen sticky top-0 z-50"> ```

### 'Hero.tsx'- 
I have implemented a new function called scrollToRepoControls that enhances the user experience by smoothly scrolling the page to the "Repo Controls" section when invoked.

And for that the button is-
>  ``` <button className="btn btn-primary w-1/2" onClick={scrollToRepoControls}>Check Trends</button>```

 ### 'Repos.tsx'-
 And repos is-
> ```<div id="repo-controls" className="flex items-center justify-between p-5 flex-col md:flex-row md:space-x-2 md:items-center mx-auto pt-20 pb-10"> ```

### Nav.tsx-
Screenshot 2024-10-14 at 01 04 43

### Hero.tsx-
Screenshot 2024-10-14 at 01 00 54

### Repos.tsx-
Screenshot 2024-10-14 at 01 06 31

### Do you want to work on this issue?

Yes

### If "yes" to above, please explain how you would technically implement this (issue will not be assigned if this is skipped)

I updated the <nav> element's class to include sticky and top-0. To further enhance navigation, I updated the functionality of the "Check Trends" button in the Hero.tsx component to smoothly scroll to the "Repos" section when clicked. Finally, I ensured that the "Repos" section has a corresponding ID in the Repos.tsx component to facilitate smooth scrolling.

### PR DESCRIPTION
**Pull request by Shrey0610**
### Summary of Changes:
I made the navbar sticky to improve accessibility as users scroll through the page.
I also fixed the "Check Trends" button so that it now scrolls smoothly to the "Repo Controls" section when clicked.

Fixes #81 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Screenshots (if applicable):
Screenshot 2024-10-14 at 23 24 04

### CODE REVIEW

File: package-lock.json
Comment (Bashamega) [inline package-lock.json:1]: Please do not update the version here

**Review by Bashamega — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (Shrey0610) [reply package-lock.json:1]: So should I generate another PR?

Comment (Bashamega) [reply package-lock.json:1]: Just push the commit to the same branch, and it will update the pr.

Comment (Shrey0610) [reply package-lock.json:1]: Ok.

Comment (Shrey0610) [reply package-lock.json:1]: I have done that please check now.

**Review by Bashamega — CHANGES_REQUESTED** [review]

File: package-lock.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "packages": {
     "": {
       "name": "trendtrack",
-      "version": "0.15.2",
+      "version": "0.15.0",
```
Comment (Bashamega) [inline package-lock.json:9]:
```suggestion
      "version": "0.15.2",
```

**Review by Bashamega — CHANGES_REQUESTED** [review]

**Review by Bashamega — APPROVED** [review]
Nice work.
I have fixed it for you.

**Pull request merged**